File: taskit/lib/features/login/domain/model/setting_model.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'setting_model.freezed.dart';

@freezed
abstract class SettingModel with _$SettingModel {
  const factory SettingModel({
    required bool isNotificationEnabled,
    required int remindBefore,
    required List<String> categories,
    required String language,
    required String theme,
  }) = _SettingModel;
}

File: taskit_backend/models/setting.model.js
import mongoose from "mongoose";
import db from "../config/db.js";

const settingSchema = new mongoose.Schema({
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    required: true,
    ref: 'user',
  },
  isNotificationEnabled: {
    type: Boolean,
    default: true,
  },
  language: {
    type: String,
    enum: ['en', 'vi', 'es'], 
    default: 'en',
  },
  theme: {
    type: String,
    enum: ['light', 'dark'],
    default: 'light',
  },
  remindBefore: {
    type: Number,
    enum: [15, 30,60,1440,10080], 
    default: 30, 
  },
},{timestamps: true});


const SettingModel = db.model("user-setting", settingSchema);
export default SettingModel;
